snuba/query/processors/tagsmap.py

```diff
@@ -136,6 +145,25 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
             if not apply_optimization:
                 return
 
+        # Do not use flattened tags if tags are being unpacked anyway. In that case
+        # using flattened tags only implies loading an additional column thus making
+        # the query heavier and slower
+        if query.get_arrayjoin_from_ast() and self.__has_tags(
+            query.get_arrayjoin_from_ast()
+        ):
+            return
+        if query.get_groupby_from_ast():
+            for expression in query.get_groupby_from_ast():
+                if self.__has_tags(expression):
+                    return
+        if query.get_having_from_ast() and self.__has_tags(query.get_having_from_ast()):
+            return
+
+        if query.get_orderby_from_ast():
+            for orderby in query.get_orderby_from_ast():
+                if self.__has_tags(orderby.node):
```

Comment:
Kind of off-topic/out of scope, but after seeing it in context, I think `orderby.expression` would make more sense than `orderby.node`.

Reply:
No strong opinion on this.